### ISSUE TITLE
NumberFormatException on get-fact

### ISSUE THREAD

**Issue by korkeala** [epiccastle/spire#36]
I have troubles getting the spire to work, simplest test is to log in to my localhost (Fedora 31 box), it throws NumberFormatException

```
spire localhost.clj 
Enter Password for user@localhost: 
Exception in thread "main" clojure.lang.ExceptionInfo: null [at line 1, column 1] {:type :sci/error, :line 1, :column 1, :message "null [at line 1, column 1]"}
	at sci.impl.utils$rethrow_with_location_of_node.invokeStatic(utils.cljc:74)
	at sci.impl.utils$rethrow_with_location_of_node.invoke(utils.cljc:58)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:481)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:461)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:511)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:494)
	at sci.impl.interpreter$eval_form.invokeStatic(interpreter.cljc:552)
	at sci.impl.interpreter$eval_form.invoke(interpreter.cljc:543)
	at sci.impl.interpreter$eval_string_STAR_.invokeStatic(interpreter.cljc:562)
	at sci.impl.interpreter$eval_string_STAR_.invoke(interpreter.cljc:557)
	at sci.impl.interpreter$eval_string$fn__2850.invoke(interpreter.cljc:574)
	at sci.impl.interpreter$eval_string.invokeStatic(interpreter.cljc:571)
	at sci.impl.interpreter$eval_string.invoke(interpreter.cljc:567)
	at sci.core$eval_string.invokeStatic(core.cljc:175)
	at sci.core$eval_string.invoke(core.cljc:138)
	at spire.core$evaluate.invokeStatic(core.clj:57)
	at spire.core$evaluate.invoke(core.clj:55)
	at spire.core$_main.invokeStatic(core.clj:96)
	at spire.core$_main.doInvoke(core.clj:73)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at spire.core.main(Unknown Source)
Caused by: java.lang.NumberFormatException: null
	at java.lang.Integer.parseInt(Integer.java:614)
	at java.lang.Integer.parseInt(Integer.java:770)
	at spire.facts$process_id_name_substring.invokeStatic(facts.clj:203)
	at spire.facts$process_id_name_substring.invoke(facts.clj:201)
	at clojure.core$mapv$fn__8450.invoke(core.clj:6914)
	at clojure.lang.PersistentVector.reduce(PersistentVector.java:343)
	at clojure.core$reduce.invokeStatic(core.clj:6829)
	at clojure.core$mapv.invokeStatic(core.clj:6905)
	at clojure.core$mapv.invoke(core.clj:6905)
	at spire.facts$process_id$fn__1577.invoke(facts.clj:212)
	at clojure.core$map$fn__5870.invoke(core.clj:2755)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.RT.seq(RT.java:535)
	at clojure.core$seq__5405.invokeStatic(core.clj:137)
	at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:24)
	at clojure.core.protocols$fn__8151.invokeStatic(protocols.clj:75)
	at clojure.core.protocols$fn__8151.invoke(protocols.clj:75)
	at clojure.core.protocols$fn__8093$G__8088__8106.invoke(protocols.clj:13)
	at clojure.core$reduce.invokeStatic(core.clj:6830)
	at clojure.core$into.invokeStatic(core.clj:6897)
	at clojure.core$into.invoke(core.clj:6889)
	at spire.facts$process_id.invokeStatic(facts.clj:214)
	at spire.facts$process_id.invoke(facts.clj:206)
	at spire.facts$fn__1587.invokeStatic(facts.clj:305)
	at spire.facts$fn__1587.invoke(facts.clj:280)
	at clojure.lang.MultiFn.invoke(MultiFn.java:229)
	at spire.facts$fetch_facts.invokeStatic(facts.clj:171)
	at spire.facts$fetch_facts.invoke(facts.clj:170)
	at spire.facts$update_facts_BANG_.invokeStatic(facts.clj:310)
	at spire.facts$update_facts_BANG_.invoke(facts.clj:309)
	at sci.impl.interpreter$fn_call.invokeStatic(interpreter.cljc:427)
	at sci.impl.interpreter$fn_call.invoke(interpreter.cljc:427)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:477)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:461)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:511)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:494)
	at sci.impl.interpreter$eval_do_STAR_$fn__2617.invoke(interpreter.cljc:384)
	at sci.impl.interpreter$eval_do_STAR_.invokeStatic(interpreter.cljc:384)
	at sci.impl.interpreter$eval_do_STAR_.invoke(interpreter.cljc:381)
	at sci.impl.interpreter$eval_do.invokeStatic(interpreter.cljc:394)
	at sci.impl.interpreter$eval_do.invoke(interpreter.cljc:391)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:431)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:429)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:468)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:461)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:511)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:494)
	at sci.impl.interpreter$eval_try.invokeStatic(interpreter.cljc:251)
	at sci.impl.interpreter$eval_try.invoke(interpreter.cljc:246)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:512)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:494)
	at sci.impl.interpreter$eval_let.invokeStatic(interpreter.cljc:83)
	at sci.impl.interpreter$eval_let.doInvoke(interpreter.cljc:59)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:435)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:429)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:468)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:461)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:511)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:494)
	at sci.impl.interpreter$eval_let.invokeStatic(interpreter.cljc:83)
	at sci.impl.interpreter$eval_let.doInvoke(interpreter.cljc:59)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:435)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:429)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:468)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:461)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:511)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:494)
	at sci.impl.interpreter$eval_do_STAR_$fn__2617.invoke(interpreter.cljc:384)
	at sci.impl.interpreter$eval_do_STAR_.invokeStatic(interpreter.cljc:384)
	at sci.impl.interpreter$eval_do_STAR_.invoke(interpreter.cljc:381)
	at sci.impl.interpreter$eval_do.invokeStatic(interpreter.cljc:394)
	at sci.impl.interpreter$eval_do.invoke(interpreter.cljc:391)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:431)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:429)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:468)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:461)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:511)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:494)
	at sci.impl.interpreter$eval_try.invokeStatic(interpreter.cljc:251)
	at sci.impl.interpreter$eval_try.invoke(interpreter.cljc:246)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:512)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:494)
	at sci.impl.interpreter$eval_let.invokeStatic(interpreter.cljc:83)
	at sci.impl.interpreter$eval_let.doInvoke(interpreter.cljc:59)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:435)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:429)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:468)
	... 18 more
```

The config file
```
# cat localhost.clj 
(ssh "user@localhost"
     (get-fact))
```
I'v tried the alpha3 and now alpha4 versions:
#spire -version
Version: 0.1.0-alpha.4

**Comment by retrogradeorbit** [comment]
I have replicated this on Fedora 31

**Comment by retrogradeorbit** [comment]
Fix has been released in 0.1.0-alpha.5. Please don't hesitate to open tickets for any other issues you may experience with Fedora.
